### PR TITLE
added DNS TXT scan module

### PR DESCRIPTION
**Pull request by WebFreak001**
**Note: DB migration should be done manually for our internal use here and the migrations be squashed**

Migration SQL:

```
ALTER TYPE _attack_attack_type ADD VALUE 'DnsTxtScan' AFTER 'DnsResolution';
ALTER TYPE _aggregationsource_source_type ADD VALUE 'DnsTxtScan' AFTER 'DnsResolution';
ALTER TYPE _searchresult_ref_type ADD VALUE 'DnsTxtScanResult' AFTER 'DnsRecordResult';
```

\+ running the auto-migrator on the here-shipped migration .toml

- includes SPF parser

TODO:

- [x] regenerate frontend
- [x] need to test again if functionality is still fully functional after rebase, possibly implement new interfaces

### CODE REVIEW

File: leech/src/modules/dns/txt.rs [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+    TxtScanInfo::HasStripeAccount,
+    TxtScanInfo::HasOneTrustSso,
+    TxtScanInfo::HasBrevoAccount,
+    TxtScanInfo::OwnsAtlassianAccounts,
+    TxtScanInfo::OwnsZoomAccounts,
+    TxtScanInfo::EmailProtonMail,
+];
+
+impl TxtScanInfo {
+    fn matcher_regex(&self) -> Option<&Regex> {
```
Comment (gammelalf) [inline leech/src/modules/dns/txt.rs:100]: The return type `Option<&'static Regex>` would be more appropriate

**Review by gammelalf — COMMENTED** [review]
Various suggestions

File: kraken/src/modules/attack_results/dns_txt_scan.rs [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+pub async fn store_dns_txt_scan_result(
+    executor: impl Executor<'_>,
+    attack_uuid: Uuid,
+    workspace_uuid: Uuid,
+    entry: DnsTxtScan,
+) -> Result<(), rorm::Error> {
+    let mut guard = executor.ensure_transaction().await?;
+    let tx = guard.get_transaction();
+
+    let user_uuid = *query!(&mut *tx, (Attack::F.started_by,))
```
[the remark this answers is not in the document]
Comment (WebFreak001) [reply kraken/src/modules/attack_results/dns_txt_scan.rs:26]: the backlog doesn't and would need to query it it seems though.

File: kraken/src/api/handler/attacks/schema.rs [hunk 3]
```diff
@@ -203,6 +203,22 @@ pub struct DnsResolutionRequest {
...
+#[derive(Deserialize, ToSchema)]
+pub struct DnsTxtScanRequest {
+    /// The leech to use
+    ///
+    /// Leave empty to use a random leech
+    pub leech_uuid: Option<Uuid>,
+
+    /// The domains to resolve
+    #[schema(value_type = Vec<String>, example = json!(["example.com", "example.org"]))]
+    pub targets: Vec<String>,
```
Comment (myOmikron) [inline kraken/src/api/handler/attacks/schema.rs:216]: really necessary to declare the value_type? `Vec<String>` should be already parsable by utoipa

File: kraken/src/rpc/definitions.rs [hunk 4]
```diff
@@ -107,4 +107,32 @@ mod impls {
...
+                        Ipv6Network::with_netmask(
+                            addr,
+                            Ipv6Addr::from([a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p]),
+                        )
+                        .unwrap(),
+                    )
+                }
+            }
+        }
+    }
```
Comment (myOmikron) [inline kraken/src/rpc/definitions.rs:137]: Should be implemented as try_from (in the leech as well as here)

**Review by myOmikron — CHANGES_REQUESTED** [review]

File: kraken/src/api/handler/attacks/schema.rs (hunk 3 above)
Comment (WebFreak001) [reply kraken/src/api/handler/attacks/schema.rs:258]: so far it has been used in every `Vec<String>` field in this file like this - should I remove it from all of them?

**Comment by WebFreak001** [comment]
see primary PR comment for migration steps, should be ready for review now.

**Pull request merged**